### PR TITLE
Stereo Features

### PR DESCRIPTION
**Pull request by mackaiver**
This processor adds some features for reconstructing multi-telescope events.
It has been tested by comparing the outputs of the various methods to the Python implementation.

Because I didn't pay proper attention (sorry) this PR also includes other things:

- new test data and telescope/camera definitions
- a bug fix when accessing the telescopes by id.

Whats still missing (don't merge this just yet):

- More docstrings
- remove unused methods
- (maybe check if results are valid on large datasets)

### CODE REVIEW

**Comment by mackaiver** [comment]
I added a bunch of comments and fixed all warnings in the code. This could be merged now after reviewing.

File: src/main/java/streams/cta/stereo/Stereo.java [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+        double[] corePosition = estimateCorePosition(planes);
+
+        data.put("stereo:estimated_direction", direction);
+        data.put("stereo:estimated_direction:x", direction[0]);
+        data.put("stereo:estimated_direction:y", direction[1]);
+        data.put("stereo:estimated_direction:z", direction[2]);
+
+        data.put("stereo:estimated_impact_position", corePosition);
+        data.put("stereo:estimated_direction:x", corePosition[0]);
+        data.put("stereo:estimated_direction:y", corePosition[1]);
```
Comment (alexeyegorov) [inline src/main/java/streams/cta/stereo/Stereo.java]: use final string variable for "stereo:estimated_direction"?

File: src/main/java/streams/cta/stereo/Stereo.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+
+                    //dont know what happens now. here is the docstring from the python
+                    // # two great circles cross each other twice (one would be
+                    // # the origin, the other one the direction of the gamma) it
+                    // # doesn't matter which we pick but it should at least be
+                    // # consistent: make sure to always take the "upper"
+                    // # solution
+                    if (product.getZ() < 0) {
+                        product = product.scalarMultiply(-1);
+                    }
```
Comment (alexeyegorov) [inline src/main/java/streams/cta/stereo/Stereo.java]: does it make sense to mark this as #TODO in order to find the meaning of this code later?

File: src/main/java/streams/cta/stereo/Stereo.java [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        final double[] telescopePosition;
+
+        Plane(int id, double phi, double theta,  Data data) {
+            this.telescopeId = id;
+            double length = (double) data.get("telescope:" + id + ":shower:length");
+            double width = (double) data.get("telescope:" + id + ":shower:width");
+            double cogX = (double) data.get("telescope:" + id + ":shower:cog:x");
+            double cogY = (double) data.get("telescope:" + id + ":shower:cog:y");
+            double psi = (double) data.get("telescope:" + id + ":shower:psi");
+            double size = (double) data.get("telescope:" + id + ":shower:size");
```
Comment (alexeyegorov) [inline src/main/java/streams/cta/stereo/Stereo.java]: use local variable for "telescope:" + id + ":shower:" ? (looks better?)

Comment (alexeyegorov) [inline src/main/java/streams/cta/stereo/Stereo.java]: XD (shower)

File: src/main/java/streams/cta/stereo/Stereo.java [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+            double[] pDirection = cameraCoordinateToDirectionVector(pX, pY, phi, theta, tel.opticalFocalLength, 0);
+            double[] cogDirection = cameraCoordinateToDirectionVector(cogX, cogY, phi, theta, tel.opticalFocalLength, 0);
+
+            this.weight = size * (length / width);
+            this.v1 = cogDirection;
+            this.v2 = pDirection;
+
+            // c  = (v1 X v2) X v1
+            Vector3D c = Vector3D.crossProduct(Vector3D.crossProduct(new Vector3D(v1), new Vector3D(v2)), new Vector3D(v1));
+            Vector3D norm = Vector3D.crossProduct(new Vector3D(v1), c);
```
Comment (alexeyegorov) [inline src/main/java/streams/cta/stereo/Stereo.java]: rename local variable "c" to "crossProduct"?

File: src/main/java/streams/cta/stereo/Stereo.java [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+     *
+     */
+    static class Plane {
+        //the tlescope id this reconstructed plane belongs to
+        final int telescopeId;
+        //the weight given to the plane
+        final double weight;
+        //two vectors describing the plane
+        final double[] v1;
+        final double[] v2;
```
Comment (alexeyegorov) [inline src/main/java/streams/cta/stereo/Stereo.java]: rename v1 and v2 to sth more self-explaining? planeVector? planeVec?

File: src/main/java/streams/cta/stereo/Stereo.java [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+         * @return a the normal vector of the plane
+         */
+        Vector3D getNormalAsVector(){
+            return new Vector3D(normalVector);
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
```
Comment (alexeyegorov) [inline src/main/java/streams/cta/stereo/Stereo.java]: personally, I don't like to inline the if-conditions. But this is just a my opinion.

File: src/main/java/streams/cta/stereo/Stereo.java (hunk 5 above)
Comment (mackaiver) [inline src/main/java/streams/cta/stereo/Stereo.java]: done

File: src/main/java/streams/cta/stereo/Stereo.java (hunk 3 above)
Comment (mackaiver) [inline src/main/java/streams/cta/stereo/Stereo.java]: done

File: src/main/java/streams/cta/stereo/Stereo.java (hunk 4 above)
Comment (mackaiver) [inline src/main/java/streams/cta/stereo/Stereo.java]: yup

File: src/main/java/streams/cta/stereo/Stereo.java (hunk 6 above)
Comment (mackaiver) [inline src/main/java/streams/cta/stereo/Stereo.java]: agreed. this code was auto generated by intellij

**Pull request merged**